db/dcrpg/pgblockchain.go

```diff
@@ -231,6 +234,7 @@ type ChainDB struct {
 	InReorg            bool
 	tpUpdatePermission map[dbtypes.TimeBasedGrouping]*trylock.Mutex
 	utxoCache          utxoStore
+	chainInfo          *dbtypes.BlockChainData
```

Comment:
Actually, having `locked_in, activated, hard_forked` in the agendas table didn't sit well with me since there it implied we were tracking the agenda item state for each tx_hash or over time or something.

Now that I look back at the agendas table, it looks almost completely redundant with the votes table.  The votes table even has vote version and vote_bits.  So it seems that all the agendas table needs is a agenda id, agenda choice, and a link to the row id in the votes table.

Reply:
Sure you make some valid points. I think we settled on a new table then to avoid too much data duplication like where a given vote tx has voted for more than one agenda. ie
```
test_dcrdata=# select tx_hash from votes where height = 173456;
                             tx_hash                              
------------------------------------------------------------------
 9e5715b7632b3bc0d12e6e301fa03ea4f6ae19a818ac6e2195b3f97dfac430a7
 f6cae37b23c6d920b506909e56a8cd8196e969cefa095acfa19bdafa53c5e26d
 f36334ccbc2a9ccb2aa2037ef0cccc49e692075b6980e5c0519b4ac36841b49f
 a8877f81af4ca83cce6dbe055a99ee9a1fdfcb88e49fa997a1ebfea172c4c279
 953a74c6c5d6f7acb207b6458da733c2c728de1e038605f250250fb19a8c09bc
(5 rows)

test_dcrdata=# select agenda_id, tx_hash from agendas where block_height = 173456;
   agenda_id    |                             tx_hash                              
----------------+------------------------------------------------------------------
 lnfeatures     | 9e5715b7632b3bc0d12e6e301fa03ea4f6ae19a818ac6e2195b3f97dfac430a7
 lnfeatures     | f6cae37b23c6d920b506909e56a8cd8196e969cefa095acfa19bdafa53c5e26d
 lnfeatures     | f36334ccbc2a9ccb2aa2037ef0cccc49e692075b6980e5c0519b4ac36841b49f
 sdiffalgorithm | a8877f81af4ca83cce6dbe055a99ee9a1fdfcb88e49fa997a1ebfea172c4c279
 lnsupport      | a8877f81af4ca83cce6dbe055a99ee9a1fdfcb88e49fa997a1ebfea172c4c279
 lnfeatures     | 953a74c6c5d6f7acb207b6458da733c2c728de1e038605f250250fb19a8c09bc
(6 rows)

```

Reply:
`lnsupport` and `sdiffalgorithm` have the same vote version.

Reply:
Maybe a table called `agenda_votes`?  I imaging there could be a high level `agendas` table for someother purpose like describing the agendas themselves, their state/outcome, etc.

Reply:
Ah, if the `agenda_votes` table has an `agenda_id` column of an integer type, the `agendas` table could map the id to the text name (e.g. "lnfeatures") along with the other high level agenda stuff.

Reply:
That high level table is in bolt db.

Reply:
Its [here](https://github.com/decred/dcrdata/tree/master/db/agendadb) infact data on `/agendas` page is got from that high level table.

Reply:
So perhaps:

```
junk=# create table agenda_votes (id SERIAL PRIMARY KEY, vote_row_id INT8, agenda_id INT8, agenda_vote_choice INT2);
CREATE TABLE

junk=# create table agendas (id SERIAL PRIMARY KEY, name TEXT, locked_in BOOLEAN, activated BOOLEAN, hard_forked BOOLEAN, description TEXT);
CREATE TABLE

junk=# \d agenda_votes
                                   Table "public.agenda_votes"
       Column       |   Type   | Collation | Nullable |                 Default                  
--------------------+----------+-----------+----------+------------------------------------------
 id                 | integer  |           | not null | nextval('agenda_votes_id_seq'::regclass)
 vote_row_id        | bigint   |           |          | 
 agenda_id          | bigint   |           |          | 
 agenda_vote_choice | smallint |           |          | 
Indexes:
    "agenda_votes_pkey" PRIMARY KEY, btree (id)

junk=# \d agendas
                               Table "public.agenda"
   Column    |  Type   | Collation | Nullable |              Default               
-------------+---------+-----------+----------+------------------------------------
 id          | integer |           | not null | nextval('agenda_id_seq'::regclass)
 name        | text    |           |          | 
 locked_in   | boolean |           |          | 
 activated   | boolean |           |          | 
 hard_forked | boolean |           |          | 
 description | text    |           |          | 
Indexes:
    "agenda_pkey" PRIMARY KEY, btree (id)

```